### PR TITLE
Solution of animal

### PR DESCRIPTION
**Pull request by svyshcho**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,36 @@
...
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Carnivore) or herbivore.hidden:
+            return
+        else:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:33]: No need to use else: because we already return in other case

File: app/main.py [hunk 2]
```diff
@@ -1 +1,36 @@
...
+    def __init__(self,
+                 name: str,
+                 health: int = 100,
+                 hidden: bool = False) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = hidden
+        Animal.alive.append(self)
+
+    def dead_animal(self) -> None:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:13]: dead_animal rename to dead. Because we already know that it related to animal

File: app/main.py [hunk 3]
```diff
@@ -1 +1,36 @@
...
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: Herbivore) -> None:
+        if isinstance(herbivore, Carnivore) or herbivore.hidden:
```
Comment (ArsenPidhoretskyi) [inline app/main.py:31]:
```suggestion
        if not isinstance(herbivore, Herbivore) or herbivore.hidden:
```
Because if someone pass an Animal?

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 3 above)
Comment (svyshcho) [reply app/main.py]: Done

File: app/main.py (hunk 2 above)
Comment (svyshcho) [reply app/main.py]: Done

File: app/main.py (hunk 1 above)
Comment (svyshcho) [reply app/main.py]: Done

**Comment by olena-hul** [comment]
Good job!

**Pull request closed without merging**